app/products/calls/simple-peer.ts

```diff
@@ -421,7 +421,12 @@ export default class Peer extends stream.Duplex {
         this.isNegotiating = true;
     }
 
-    _destroy(err: Error | null, cb: (error: Error | null) => void) {
+    destroy(err?: Error, cb?: (error: Error | null) => void): this {
+        this._destroy(err, cb);
+        return this;
+    }
```

Comment:
Interestingly the original implementation has this comment on top of  [`destroy()`](https://github.com/feross/simple-peer/blob/master/index.js#L427-L432):

```
  // TODO: Delete this method once readable-stream is updated to contain a default
  // implementation of destroy() that automatically calls _destroy()
  // See: https://github.com/nodejs/readable-stream/issues/283
```

This may very well be the reason it was removed. As you correctly found out `_destroy` is being automatically called by the underlying readable-stream implementation:

https://github.com/nodejs/readable-stream/blob/97fd94f404709542a7d69b271eddfcc34345916e/lib/internal/streams/destroy.js#L35

I am wondering if there's some race/timing issue going on here which is causing this.

Reply:
Yep, I think that's probably what was happening -- it's a race condition, because I think `_destroy` is called "eventually", when we needed it to be called right when we want it to be.